services/roomService/src/requestHandlers/CoveyTownRequestHandlers.ts

```diff
@@ -233,3 +261,65 @@ export function townSubscriptionHandler(socket: Socket): void {
     townController.updatePlayerLocation(s.player, movementData);
   });
 }
+
+export async function mazeTimeHandler(): Promise<ResponseEnvelope<MazeCompletionTimeListResponse>> {
+  try {
+    const results = await pool.query(getMazeCompletionTime);
+    return {
+      isOK: true,
+      response: {
+        mazeCompletionTimes: results.rows.map(value => ({
+          playerID: value.player_id,
+          username: value.username,
+          time: value.time,
+        })),
+      },
+    };
+  } catch (error) {
+    return {
+      isOK: false,
+    };
+  }
+}
+
+export async function mazeTimeCreateHandler(
+  requestData: MazeCompletionTimeCreateRequest,
+): Promise<ResponseEnvelope<Record<string, null>>> {
+  const { coveyTownID, playerID, username, time } = requestData;
+
+  if (!coveyTownID || !playerID || !username || !time) {
+    return {
+      isOK: false,
+      message: 'Please include a coveyTownID, playerID, username, and time in seconds',
+    };
+  }
+
+  const townStore = CoveyTownsStore.getInstance();
+  const townController = townStore.getControllerForTown(coveyTownID);
+
+  if (!townController) {
+    return {
+      isOK: false,
+      message: 'Please include a valid coveyTownID',
+    };
+  }
+
+  if (!townController.hasPlayer(playerID)) {
+    return {
+      isOK: false,
+      message: 'Player does not exist in specified town',
+    };
+  }
+
+  try {
+    await pool.query(insertMazeCompletionTime, [playerID, username, time]);
```

Comment:
If a player plays multiple times and gets a better score, will this update their previous score or be added to the database as a separate entry?

Reply:
I was under the impression we were planning on doing arcade style where new scores from the same username/id pair will add a new row instead of updating ever (esp since usernames are not unique and ids are not persisted)

Reply:
Ok yeah, that'll probably be easier to keep track of too.  Also wondering if we have to worry about the size of the database?  It might be unnecessary to keep entries that have times so slow we wouldn't even be able to add them to the leaderboard.

Reply:
Discussed offline: we agreed that the database should hold all information and we will query with the subset of information as needed.